rinja/src/filters/alloc.rs

```diff
@@ -0,0 +1,657 @@
+/// }
+///
+/// assert_eq!(
+///     Example { example: "hello" }.to_string(),
+///     "<div>Hello</div>"
+/// );
+///
+/// assert_eq!(
+///     Example { example: "hElLO" }.to_string(),
+///     "<div>Hello</div>"
+/// );
+/// # }
+/// ```
+#[inline]
+pub fn capitalize(s: impl fmt::Display) -> Result<String, fmt::Error> {
+    fn capitalize(s: &str) -> Result<String, fmt::Error> {
+        let mut chars = s.chars();
+        if let Some(c) = chars.next() {
+            let mut replacement = String::with_capacity(s.len());
+            replacement.extend(c.to_uppercase());
+            replacement.push_str(&chars.as_str().to_lowercase());
+            Ok(replacement)
+        } else {
+            Ok(String::new())
+        }
+    }
+
+    let mut buffer;
+    capitalize(try_to_str!(s => buffer))
+}
+
+/// Count the words in that string.
+///
+/// ```
+/// # #[cfg(feature = "code-in-doc")] {
+/// # use rinja::Template;
+/// /// ```jinja
+/// /// <div>{{ example|wordcount }}</div>
+/// /// ```
+/// #[derive(Template)]
+/// #[template(ext = "html", in_doc = true)]
+/// struct Example<'a> {
+///     example: &'a str,
+/// }
+///
+/// assert_eq!(
+///     Example { example: "rinja is sort of cool" }.to_string(),
+///     "<div>5</div>"
+/// );
+/// # }
+/// ```
+pub fn wordcount(s: impl fmt::Display) -> Result<usize, fmt::Error> {
+    let mut buffer;
+    Ok(try_to_str!(s => buffer).split_whitespace().count())
+}
+
+/// Return a title cased version of the value. Words will start with uppercase letters, all
+/// remaining characters are lowercase.
+///
+/// ```
+/// # #[cfg(feature = "code-in-doc")] {
+/// # use rinja::Template;
+/// /// ```jinja
+/// /// <div>{{ example|title }}</div>
+/// /// ```
+/// #[derive(Template)]
+/// #[template(ext = "html", in_doc = true)]
+/// struct Example<'a> {
+///     example: &'a str,
+/// }
+///
+/// assert_eq!(
+///     Example { example: "hello WORLD" }.to_string(),
+///     "<div>Hello World</div>"
+/// );
+/// # }
+/// ```
+pub fn title(s: impl fmt::Display) -> Result<String, fmt::Error> {
+    let mut buffer;
+    let s = try_to_str!(s => buffer);
+    let mut need_capitalization = true;
+
+    // Sadly enough, we can't mutate a string when iterating over its chars, likely because it could
+    // change the size of a char, "breaking" the char indices.
+    let mut output = String::with_capacity(s.len());
+    for c in s.chars() {
+        if c.is_whitespace() {
+            output.push(c);
+            need_capitalization = true;
+        } else if need_capitalization {
+            match c.is_uppercase() {
+                true => output.push(c),
+                false => output.extend(c.to_uppercase()),
+            }
+            need_capitalization = false;
+        } else {
+            match c.is_lowercase() {
+                true => output.push(c),
+                false => output.extend(c.to_lowercase()),
+            }
+        }
+    }
+    Ok(output)
+}
+
+#[cfg(all(test, feature = "alloc"))]
```

Comment:
I think it can just be:

```suggestion
#[cfg(test)]
```

since this file is already imported only with `#[cfg(feature = "alloc")]`.

Reply:
Ah, yes. Replaced.